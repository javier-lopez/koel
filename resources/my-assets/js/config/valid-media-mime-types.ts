export const validMediaMimeTypes = [
  'audio/mp3',
  'audio/mpeg',
  'audio/ogg',
  'audio/x-flac',
  'audio/x-aac'
]
