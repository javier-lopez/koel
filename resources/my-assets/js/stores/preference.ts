import { userStore } from '.'
import { ls } from '@/services'

export const preferenceStore = {
  storeKey: '',

  state: {
    volume: 7,
    notify: true,
    repeatMode: 'NO_REPEAT',
    showExtraPanel: true,
    confirmClosing: false,
    equalizer: {
      preamp: 0,
      gains: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
    },
    artistsViewMode: null,
    albumsViewMode: null,
    selectedPreset: -1,
    transcodeOnMobile: false,
    supportBarNoBugging: false,
    showAlbumArtOverlay: true
  } as { [key: string]: any },

  init (user?: User): void {
    const initUser = user || userStore.current
    this.storeKey = `preferences_${initUser.id}`
    this.state = Object.assign(this.state, ls.get(this.storeKey, this.state))
    this.setupProxy()
  },

  /**
   * Proxy the state properties, so that each can be directly accessed using the key.
   */
  setupProxy (): void {
    Object.keys(this.state).forEach((key: string): void => {
      Object.defineProperty(this, key, {
        get: (): any => this.get(key),
        set: (value: any): void => this.set(key, value),
        configurable: true
      })
    })
  },

  set (key: string, val: any): void {
    this.state[key] = val
    this.save()
  },

  get (key: string): any {
    return key in this.state ? this.state[key] : null
  },

  save (): void {
    ls.set(this.storeKey, this.state)
  }
} as { [key: string]: any }
