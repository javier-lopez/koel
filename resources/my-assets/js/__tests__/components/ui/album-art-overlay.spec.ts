import flushPromises from 'flush-promises'
import factory from '@/__tests__/factory'
import Component from '@/components/ui/album-art-overlay.vue'
import { albumStore } from '@/stores/album'
import { shallow } from '@/__tests__/adapter'
import { mock } from '@/__tests__/__helpers__'
import { eventBus } from '@/utils'
import { events } from '@/config'
import { preferenceStore } from '@/stores'

describe('components/ui/album-art-overlay', () => {
  afterEach(() => {
    jest.resetModules()
    jest.clearAllMocks()
  })

  it('requests album thumbnail when a song starts playing', async () => {
    preferenceStore.state.showAlbumArtOverlay = true
    const wrapper = shallow(Component)
    const song = factory<Song>('song')
    const getCoverThumbnailMock = mock(albumStore, 'getThumbnail').mockResolvedValue('http://localhost/foo_thumb.jpg')
    eventBus.emit(events.SONG_STARTED, song)
    await flushPromises()
    expect(getCoverThumbnailMock).toHaveBeenCalledWith(song.album)
    expect(wrapper).toMatchSnapshot()
  })

  it('doesnt request album thumbnail if user preference is set to not displaying album art overlay', async () => {
    preferenceStore.state.showAlbumArtOverlay = false
    shallow(Component)
    const song = factory<Song>('song')
    const getCoverThumbnailMock = mock(albumStore, 'getThumbnail')
    eventBus.emit(events.SONG_STARTED, song)
    await flushPromises()
    expect(getCoverThumbnailMock).not.toHaveBeenCalled()
  })
})
