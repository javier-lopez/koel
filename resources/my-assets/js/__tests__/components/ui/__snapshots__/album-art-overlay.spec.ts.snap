// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ui/album-art-overlay requests album thumbnail when a song starts playing 1`] = `<div style="background-image: url(http://localhost/foo_thumb.jpg);"></div>`;
